### PR TITLE
MM-18383: updated tests using fatal to use assert

### LINKED ISSUE #12098: Migrate tests from "app/command_groupmsg_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `app/command_groupmsg_test.go` file.

The expected way to migrate this is to go to the `app/command_groupmsg_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-18383

### PR DESCRIPTION
**Pull request by renilJoseph**
**Summary**
Migrate tests from "app/command_groupmsg_test.go" to use testify

**Ticket Link**
https://mattermost.atlassian.net/browse/MM-18383

Fixes #12098

### CODE REVIEW

File: app/command_groupmsg_test.go [hunk 1]
```diff
@@ -10,35 +10,44 @@ import (
...
-	}
-
-	if users, _ := groupMsgUsernames("    test1,       test2,,123,@321,+123"); len(users) != 5 || users[0] != "test1" || users[1] != "test2" || users[2] != "123" || users[3] != "321" || users[4] != "+123" {
-		t.Fatal("error parsing different types of users")
-	}
+	assert := assert.New(t)
+
+	users, parsedMessage := groupMsgUsernames("")
+	assert.Len(users, 0)
+	assert.Equal(parsedMessage, "", "error parsing empty message")
```
Comment (mgdelacroix) [inline app/command_groupmsg_test.go]: In this case, we can use directly the `assert.Empty` function, without error message

File: app/command_groupmsg_test.go [hunk 2]
```diff
@@ -10,35 +10,44 @@ import (
...
+
+	users, parsedMessage = groupMsgUsernames("test")
+	assert.Len(users, 1)
+	assert.Equal(parsedMessage, "", "error parsing simple user")
```
Comment (mgdelacroix) [inline app/command_groupmsg_test.go]: same here

File: app/command_groupmsg_test.go [hunk 3]
```diff
@@ -10,35 +10,44 @@ import (
...
+
+	users, parsedMessage = groupMsgUsernames("test1, test2, test3 , test4")
+	assert.Len(users, 4)
+	assert.Equal(parsedMessage, "", "error parsing various users")
```
Comment (mgdelacroix) [inline app/command_groupmsg_test.go]: same here

File: app/command_groupmsg_test.go [hunk 4]
```diff
@@ -10,35 +10,44 @@ import (
...
+	assert.Len(users, 2)
+	assert.Equal(parsedMessage, "message with spaces", "error parsing message")
+
+	users, parsedMessage = groupMsgUsernames("test1, test2 message with, comma")
+	assert.Len(users, 2)
+	assert.Equal(parsedMessage, "message with, comma", "error parsing messages with comma")
+
+	users, parsedMessage = groupMsgUsernames("test1,,,test2")
+	assert.Len(users, 2)
+	assert.Equal(parsedMessage, "", "error parsing multiple commas in username")
```
Comment (mgdelacroix) [inline app/command_groupmsg_test.go]: same here

File: app/command_groupmsg_test.go [hunk 5]
```diff
@@ -10,35 +10,44 @@ import (
...
+	assert.Equal(parsedMessage, "other message", "error parsing strange usage of spaces")
+
+	users, _ = groupMsgUsernames("    test1,       test2,,123,@321,+123")
+	assert.Len(users, 5)
+	assert.Equal(users[0], "test1")
+	assert.Equal(users[1], "test2")
+	assert.Equal(users[2], "123")
+	assert.Equal(users[3], "321")
+	assert.Equal(users[4], "+123")
+	assert.Equal(parsedMessage, "other message", "error parsing strange usage of spaces")
```
Comment (mgdelacroix) [inline app/command_groupmsg_test.go]: This line looks like it's [repeated from the test just before](https://github.com/mattermost/mattermost-server/pull/12175/files#diff-54e30e28064feaf5d69d3e3612115290R43) and not belonging to this one, is that right?

Comment (renilJoseph) [reply app/command_groupmsg_test.go]: yes, Sorry for that mistake.

Comment (mgdelacroix) [reply app/command_groupmsg_test.go]: No worries! Thanks for the fixes!! :)

Comment (renilJoseph) [reply app/command_groupmsg_test.go]: @mgdelacroix hi, anything else to do from my side ?

**Pull request merged**